inst/rstudio/templates/project/resources/common/04-citations.Rmd

```diff
@@ -0,0 +1,20 @@
+# Footnotes and citations 
+
+## Footnotes
+
+Footnotes are put inside the square brackets after a caret `^[]`. Like this one ^[This is a footnote.]. 
+
+## Citations
+
+Reference items in your bibliography file(s) using `@key`.
+
+For example, we are using the **bookdown** package [@R-bookdown] (check out the last code chunk in index.Rmd to see how this citation key was added) in this sample book, which was built on top of R Markdown and **knitr** [@xie2015] (this citation was added manually in an external file book.bib). 
+Note that the `.bib` files need to be listed in the index.Rmd with the YAML `bibliography` key.
+
+<!--bs4_book:start-->
+
+The `bs4_book` theme makes footnotes appear inline when you click on them. In this example book, we added `csl: chicago-fullnote-bibliography.csl` to the `index.Rmd` YAML, and include the `.csl` file.
```

Comment:
Is this `.csl` file mandatory to `bs4_book`? I mean, does this feature work without a `.csl` file, and does it work with other `.csl` files?

Reply:
The CSL is not mandatory. The feature is : 

* If citation are in footnote style, then the footnotes are transformed to be show inline 
  ![image](https://user-images.githubusercontent.com/6791940/126756825-c5138d42-160f-460d-b62c-142e75b0cb69.png)
* If not, then they appear as usual. This is without CSL:
  ![image](https://user-images.githubusercontent.com/6791940/126756911-232f4110-1fe4-4634-98b9-17758ff3c389.png)
  but currently, the `split_bib` option is hard coded to FALSE (should be opt-in #1185)

So CSL style is mandatory to make the citation appear inline in clickable footnote upperscript number. But this seems ok to do that only if the citation style allows it.

Also, this feature will work with any citation style making them footnotes, so ones from https://www.zotero.org/styles?format=note I guess, but I don't know other popular unless this one. For example, it works also with `the-journal-of-clinical-ethics.csl` (https://www.zotero.org/styles/the-journal-of-clinical-ethics)

Does it answer your questions ?

Reply:
Yes. I prefer not shipping a csl file with this package unless necessary, since shipping it means we'll need to maintain it (although I guess it requires little maintenance).

I'll let you decide.

Reply:
I think it is important to show this feature that is not common and difficult to guess. 
I see several soluton: 
* We can use a URL and it would work. `csl` field supports that. 
````yaml
csl: https://www.zotero.org/styles/chicago-fullnote-bibliography
````
* We can have a script to maintain external data up-to-date by downloading the source. Either https://www.zotero.org/styles/chicago-fullnote-bibliography or directly for repo https://github.com/citation-style-language/styles/blob/master/chicago-fullnote-bibliography.csl

Maybe the first solution is best ?

Reply:
The downside of the first solution is that users won't be able to build the book offline. One intermediate solution is that we download it automatically the first time when the book is built, i.e., we add a code chunk to download the csl if it doesn't exist:

````
```{r, download-csl, include=FALSE}
if (!file.exists("chicago-fullnote-bibliography.csl")) xfun::download_file(
  "https://www.zotero.org/styles/chicago-fullnote-bibliography",
  "chicago-fullnote-bibliography.csl"
)
```
````

Reply:
Oh this is in fact a nice example to show! I like that.

Reply:
Will this be in the template book itself? Or can the template function run this code upon book creation? I think providing that code chunk in the template feels like ["inessential weirdness"](https://www.harihareswara.net/sumana/2016/05/21/0). I worry it looks complicated and will alienate users. On the other hand, the inclusion of this file seems integral to how bs4_book works. 

Would you be able to live with including this file, but then in that section of the template me noting that we suggest downloading an up-to-date CSL file from https://www.zotero.org/styles/?

That feels like a very useful link to share, as does a real working example inside the template itself.

Reply:
* Add the download function to `create_bs4_book()` and make it work offline (do we have a way to detect offline usage ?)
* Add a not in the template about CSL
* Add a link to zotero website in `?bs4_book()`

Reply:
I have added a function to try download asset. When offline (or rather download is not working), it will print a message

````r
> path = tempfile()
> create_bs4_book(path)
>>> ‘chicago-fullnote-bibliography.csl’ can't be downloaded. Please retrieve it manually at ‘https://www.zotero.org/styles/chicago-fullnote-bibliography’ or remove its usage from the template.
[1] "C:\\Users\\chris\\AppData\\Local\\Temp\\Rtmp2XNeZd\\file18ca04f252c6d"
```

Best way would be to use `curl::has_internet()` / `curl::nslookup()` to test offline or not, but we don't have curl in dependency and I don't want to add it for this only.

With 3a0441e we now have the csl file downloaded when the template is created the first time.